src/services/roads/index.ts

```diff
@@ -0,0 +1,44 @@
+const _ = require('lodash');
+import { computeDestinationPoint } from 'geolib';
+const query_overpass = require('query-overpass');
+
+function getBBoxCoordinates(lat, lon, radius) {
+    let points = [];
+    points.push(computeDestinationPoint({lat,lon},radius, 225));
+    points.push(computeDestinationPoint({lat,lon},radius, 45));
+    return points;
+}
+
+function removeDuplicates(array) {
+    let newArray = [];
+    array.map((street) => {
+        if(newArray.filter((s) => newArray.length > 0 ?
+                s.includes(_.last(street.split(' '))) : true
+        ).length===0) {
+            newArray.push(street);
+        }
+    });
+    return newArray;
+}
+
+export default async function getNearestRoads(req) {
+    const { station, radius } = req.body;
+    const bbox = getBBoxCoordinates(station[0], station[1], radius);
+    const query = `[out:json];(node["highway"~"tertiary|residential"](${bbox[0].latitude},${bbox[0].longitude},${bbox[1].latitude},${bbox[1].longitude});way["highway"~"tertiary|residential"](${bbox[0].latitude},${bbox[0].longitude},${bbox[1].latitude},${bbox[1].longitude});relation["highway"~"tertiary|residential"](${bbox[0].latitude},${bbox[0].longitude},${bbox[1].latitude},${bbox[1].longitude}););(._;>;);out;`;
+
+    return new Promise((resolve, reject) => {
+        query_overpass(query, (err, res) => {
+            if(res) {
+                const streets = res.features.filter((feature) => {
+                    const {properties: {tags: {name}}} = feature;
+                    return !!name;
```

Comment:
These !! are not needed.

Reply:
I don't see any benefit from destructuring here maybe just write feature.properties.tags.name ?